hoomd/md/pair/alch.py

```diff
@@ -0,0 +1,220 @@
+        superclasses += (_AlchemicalMethods,)
+        return super().__new__(cls, name, superclasses, attributedict)
+
+    def __init__(cls, name, superclasses, attributedict):
+        cls._reserved_default_attrs['_alchemical_parameters'] = list
+        cls._accepted_modes = ('none', 'shift')
+        super().__init__(name, superclasses, attributedict)
+
+
+# Perhaps define a true base alch particle to base this off of just adding the
+# MD components mirroring the cpp side
+class AlchemicalMDParticle(_HOOMDBaseObject):
+    __slots__ = ['force', 'name', 'typepair', '_cpp_obj', '_mass']
+
+    def __new__(cls,
+                force: _AlchemicalPairPotential,
+                name: str = '',
+                typepair: tuple = None,
+                mass: float = 1.0):
+        typepair = tuple(sorted(typepair))
+        if (typepair, name) in force.alchemical_particles:
+            return force.alchemical_particles[typepair, name]
+        return super().__new__(cls)
+
+    def __init__(self,
+                 force: _AlchemicalPairPotential,
+                 name: str = '',
+                 typepair: tuple = None,
+                 mass: float = 1.0):
+        self.force = force
+        self.name = name
+        self.typepair = typepair
+        self.force._add_dependent(self)
+        self._mass = mass
+        if self.force._attached:
+            self._attach()
+
+    def _attach(self):
+        self._cpp_obj = self.force._cpp_obj.getAlchemicalPairParticle(
+            *map(self.force._simulation.state.particle_types.index,
+                 self.typepair),
+            self.force._alchemical_parameters.index(self.name))
+        self.mass = self._mass
+        self._mass = self.mass
+
+    # Need to enable and disable via synced list of alchemostat
+    def _add(self, simulation):
+        self._enable()
+        super()._add(simulation)
+
+    @log(default=False)
+    def mass(self):
+        if self._attached:
+            return self._cpp_obj.mass
+        else:
+            return self._mass
+
+    @mass.setter
+    def mass(self, M):
+        if self._attached:
+            self._cpp_obj.mass = M
+        else:
+            self._mass = M
+
+    @log
+    def value(self):
+        return self.force.params[self.typepair][self.name] * (
+            self._cpp_obj.alpha if self._attached else 1.)
+
+    @log(default=False, requires_run=True)
+    def avalue(self):
+        return self._cpp_obj.alpha
+
+    @log(default=False, requires_run=True)
+    def amomentum(self):
+        return self._cpp_obj.alpha
+
+    @log(default=False, requires_run=True)
+    def mu(self):
+        return self._cpp_obj.mu
+
+    @log(default=False, requires_run=True, category='particle')
+    def aforces(self):
+        return self._cpp_obj.forces
+
+    @log(requires_run=True)
+    def net_aforce(self):
+        return self._cpp_obj.net_force()
+
+    def _enable(self):
+        self.force._cpp_obj.enableAlchemicalPairParticle(self._cpp_obj)
```

Comment:
This should only be called when we're adding the alchemical particle to an alchemostat method. And removing it should probably disable it and set the parameter value in the dict to it's current value. That hasn't been implemented yet however.